### PR TITLE
feat(GraphQL): Add GraphQL filter ability for node name from set

### PR DESCRIPTION
**Pull request by aali309**
# Welcome to Cryostat! 👋
## Before contributing, make sure you have:
* [x] Read the [contributing guidelines](https://github.com/cryostatio/cryostat/blob/main/CONTRIBUTING.md)
* [x] Linked a relevant issue which this PR resolves
* [x] Linked any other relevant issues, PR's, or documentation, if any
* [x] Resolved all conflicts, if any
* [x] Rebased your branch PR on top of the latest upstream `main` branch
* [x] Attached at least one of the following labels to the PR: `[chore, ci, docs, feat, fix, test]`
* [x] Signed the last commit: `git commit --amend --signoff`
_______________________________________________

Fixes: #1433

## Description of the change:
*This change adds allows filtering of multiple names instead of just one name*

## Motivation for the change:
*This change is helpful because one can now filter multiple matching names with just one request instead of filtering one name at time for each request*

## How to manually test:
1. *Run CRYOSTAT_IMAGE=quay.io... sh smoketest.sh...*
2. * mvn clean verify *
3. * mvn clean verify -Dit=tests=GraphQLIT.java*

### CODE REVIEW

**Comment by andrewazores** [comment]
If the goal of closing and reopening the PR is to get the tests to re-run, please just ping one of @cryostatio/reviewers and we can manually re-trigger the tests. Closing the PR will cause the automation to move it to the Done column on the project board, it might cause other bot automation to re-label it, more CI checks are re-done than need to be, etc.

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+    "java.compile.nullAnalysis.mode": "automatic",
+    "java.configuration.updateBuildConfiguration": "interactive"
```
Comment (andrewazores) [inline .vscode/settings.json:3]: Please remove this file

File: pom.xml [hunk 2]
```diff
@@ -661,7 +667,9 @@
               <argument>5m</argument>
               <argument>sh</argument>
               <argument>-c</argument>
+
               <argument>until podman ps --filter health=healthy --format '{{.Names}} {{.Status}}' | grep ${cryostat.itest.grafana.containerName}; do sleep 5; done</argument>
+
```
Comment (andrewazores) [inline pom.xml:672]: Could you undo the changes to this file please?

File: src/main/java/io/cryostat/net/web/http/api/v2/graph/TargetNodesFetcher.java [hunk 3]
```diff
@@ -110,6 +110,13 @@ public List<TargetNode> getAuthenticated(DataFetchingEnvironment environment) th
                             .filter(n -> Objects.equals(n.getName(), nodeName))
                             .collect(Collectors.toList());
         }
+        if (filter.contains(FilterInput.Key.NAMES)) {
+            List<String> names = filter.get(FilterInput.Key.NAMES);
+            result =
+                    result.stream()
+                            .filter(n -> names.contains(n.getName()))
```
Comment (andrewazores) [inline src/main/java/io/cryostat/net/web/http/api/v2/graph/TargetNodesFetcher.java:117]: There is another possibility here that I was considering.

Rather than having the `names` filter be a list of exact string matches that the given node's name must be a member of, this could reuse the [LabelSelectorMatcher](https://github.com/cryostatio/cryostat/blob/main/src/main/java/io/cryostat/net/web/http/api/v2/graph/labels/LabelSelectorMatcher.java). I will give a bit more thought to which approach I would prefer we go for here. Regardless, I suggest you take a look at that class and its unit tests and familiarize yourself with it.

File: web-client
Comment (andrewazores) [inline web-client:1]: Please undo changes to this submodule.

**Review by andrewazores — CHANGES_REQUESTED** [review]

File: .vscode/settings.json (hunk 1 above)
Comment (aali309) [reply .vscode/settings.json:3]: done

File: src/main/java/io/cryostat/net/web/http/api/v2/graph/TargetNodesFetcher.java (hunk 3 above)
Comment (andrewazores) [reply src/main/java/io/cryostat/net/web/http/api/v2/graph/TargetNodesFetcher.java:117]: Maybe a nice flexible way to do this would be:

1. Accept the `name` filter for a single exact match. This is the same thing as `names: ['one-match']` and already done.
2. Accept the `names` filter for a list of exact matches. This is what you have done here.
3. Accept a `nameMatch` filter that accepts the `LabelSelectorMatcher` syntax.

File: web-client
Comment (andrewazores) [reply web-client:1]: I don't know if that will help.

I suggest you do `git log web-client` to see what commits have been made to the submodule. You should only see log entries like this:

```
commit feeae5042b194f4ea6f2ceea794731e58a9034e6
Author: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Date:   Thu Jun 8 19:50:58 2023 +0000

    build(web-client): update submodule to db0d4cf

commit 11bfbf6728814f5b6a7e8af55010c86131af9c50
Author: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Date:   Tue Jun 6 19:36:04 2023 +0000

    build(web-client): update submodule to 4889b2b

commit 7e3f5fcf53638cbb8b82e8cf71ac4a52c6b6cced
Author: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Date:   Tue Jun 6 19:22:41 2023 +0000

    build(web-client): update submodule to 0dd044c
```

Since you update the submodule in this PR, there will be an additional commit that you authored in the log. If that commit only updated the submodule then you may simply `git revert abcd1234` the specific commit. You can check what changes were made by that commit with `git show abcd1234`. If the commit made other changes to other files as well, you can try `git revert --no-commit abcd1234` to undo the patch, then use `git add` and `git reset` to stage/unstage changes so that your changes are all still staged except for the `web-client` update. Then you can re-commit the changes and force-push, and do `git submodule update` to reset the `web-client` submodule.

File: src/main/java/io/cryostat/net/web/http/api/v2/graph/TargetNodesFetcher.java (hunk 3 above)
Comment (aali309) [reply src/main/java/io/cryostat/net/web/http/api/v2/graph/TargetNodesFetcher.java:117]: for now do i need to do any changes?

File: .vscode/settings.json (hunk 1 above)
Comment (aali309) [reply .vscode/settings.json]: i run --> ```git rm --cached ~/Desktop/workStation/cryostat/.vscode/settings.json ``` the way you showed me the other day and then added this on my .gitconfig
``` [user]
        email = atali@redhat.com
        name = Atif Ali
        signingkey = 44ECBBDDF9A888E0
[commit]
        gpgsign = true
[core]
        excludesFile = ~/.gitignore
```
and i have that on my .gitignore 
```
 .vscode
*.swp
``` 
but still it added the file.. did i do any thing wrong on this set up?

Comment (andrewazores) [reply .vscode/settings.json]: It's probably case sensitive. I have `~/.gitconfig` contents:

```
...
[core]
    autocrlf = input
    excludesfile = ~/.gitignore
...
```

**Review by andrewazores — CHANGES_REQUESTED** [review]

File: web-client
[the remark this answers is not in the document]
Comment (aali309) [reply web-client:1]: i must have done the revert in a wrong way

[the remark this answers is not in the document]
Comment (aali309) [reply web-client:1]: please check again... I reverted a different commit. I hope it fixes it

[the remark this answers is not in the document]
Comment (andrewazores) [reply web-client:1]: It looks like the web-client and pom.xml are still modified :(

File: src/main/resources/types.graphqls [hunk 4]
```diff
@@ -17,6 +17,7 @@ input EnvironmentNodeFilterInput {
 input TargetNodesFilterInput {
     id: Int
     name: String
+    names: [String]
```
Comment (andrewazores) [inline src/main/resources/types.graphqls:20]: Change looks good! Could you also add this `names: [String]` filter to the other filter inputs? `EnvironmentNodeFilterInput`, `DescendantTargetsFilterInput`, `ActiveRecordingFilterInput`, and `ArchivedRecordingFilterInput`. I think the implementation will be basically just a copy-paste job across a few other files, and the tests should all be very similar as well.

Comment (aali309) [reply src/main/resources/types.graphqls:20]: will sure do

**Comment by andrewazores** [comment]
Please don't edit the pull request labels to manipulate it into triggering CI runs, either... if you need the tests run, you can either run them on your development machine, or ask a reviewer to trigger a re-run.

File: src/main/java/io/cryostat/net/web/http/api/v2/graph/TargetNodesFetcher.java (hunk 3 above)
Comment (andrewazores) [reply src/main/java/io/cryostat/net/web/http/api/v2/graph/TargetNodesFetcher.java:117]: https://github.com/cryostatio/cryostat/issues/1552

File: src/main/java/io/cryostat/net/web/http/api/v2/graph/ActiveRecordingsFetcher.java [hunk 5]
```diff
@@ -93,11 +93,20 @@ public Active getAuthenticated(DataFetchingEnvironment environment) throws Excep
...
 
         FilterInput filter = FilterInput.from(environment);
+        List<String> names = new ArrayList<>();
         if (filter.contains(FilterInput.Key.NAME)) {
             String recordingName = filter.get(FilterInput.Key.NAME);
+            names.add(recordingName);
+        }
+        if (filter.contains(FilterInput.Key.NAMES)) {
+            List<String> recordingNames = filter.get(FilterInput.Key.NAMES);
+            names.addAll(recordingNames);
```
Comment (andrewazores) [inline src/main/java/io/cryostat/net/web/http/api/v2/graph/ActiveRecordingsFetcher.java:103]: This implementation is going to behave differently from the others:

```
var names = [];
if (filters.contains(NAME_FILTER))
  names.add(filters.get(NAME_FILTER))
if (filters.containers(NAMES_FILTER))
  names.addAll(filters.get(NAMES_FILTER))
recordings = recordings.filter(r -> names.contains(r.getName())) // if r.getName() matches the NAME or any of the NAMES
```

vs

```
if (filters.contains(NAME_FILTER))
  recordings = recordings.filter(r -> r.getName().equals(filters.get(NAME_FILTER))) // if r.getName() matches the one NAME filter
if (filters.contains(NAMES_FILTER))
  recordings = recordings.filter(r -> filters.get(NAMES_FILTER)).contains(r.getName()) // if r.getName() matches any of the NAMES
```

If the client specifies both the `name: ""` and `names: []` filter inputs, we basically have that the first implementation does a set union and the second does a set intersection.

Comment (aali309) [reply src/main/java/io/cryostat/net/web/http/api/v2/graph/ActiveRecordingsFetcher.java:103]: Should we have both .matches and .
contains implemented?

Comment (andrewazores) [reply src/main/java/io/cryostat/net/web/http/api/v2/graph/ActiveRecordingsFetcher.java:103]: Those specific method calls are an internal detail that clients don't need to worry about. What matters is that, regardless of which type of resource they are querying for, the behaviour they get when specifying a filter with both `NAME` and `NAMES` should be consistent. It should only be done one way or the other.

Generally in cases like this, I like to follow the "principle of least surprise." A user might come across this API and not realize that there are both `NAME` and `NAMES` filters, and accidentally use one, or the other, or both. If they do that, then that behaviour should be consistent with how any other pair of filters combines.

If a user queries for active recordings and specifies a name filter as well as a label filter, what result do they get? Is it all recordings that have that name or those labels, or is it all recordings that have both that name and those labels?

File: src/test/java/itest/GraphQLIT.java [hunk 6]
```diff
@@ -588,6 +602,392 @@ void testNodesHaveIds() throws Exception {
...
+        target2.name = "service:jmx:rmi:///jndi/rmi://cryostat-itests:9093/jmxrmi";
+        target2.nodeType = "JVM";
+
+        assertThat(targetNodes, hasItem(target1));
+        assertThat(targetNodes, hasItem(target2));
+    }
+
+    @Test
+    @Order(10)
+    public void testQueryForFilteredActiveRecordingsByNames() throws Exception {
```
Comment (andrewazores) [inline src/test/java/itest/GraphQLIT.java:648]: Shouldn't this test be calling the `/api/v2.2/graphql` endpoint at some point? I don't think it is currently exercising the active recording "names" filter.

File: src/test/java/itest/GraphQLIT.java [hunk 7]
```diff
@@ -588,6 +602,392 @@ void testNodesHaveIds() throws Exception {
...
+
+        MatcherAssert.assertThat(
+                "list should have size 0 after deleting recordings",
+                listResp.size(),
+                Matchers.equalTo(0));
+    }
+
+    @Test
+    @Order(11)
+    public void shouldReturnArchivedRecordingsFilteredByNames() throws Exception {
```
Comment (andrewazores) [inline src/test/java/itest/GraphQLIT.java:789]: Same here, this test exercises a recording creation/archive/deletion workflow, but it doesn't actually use GraphQL to do so.

Comment (aali309) [reply src/test/java/itest/GraphQLIT.java:789]: I  saw on the  HTTP_API.md  GET/api/vi/recordings will return all archived recordings then I can just match the names?

Comment (aali309) [reply src/test/java/itest/GraphQLIT.java:789]: i got it though... Also saw an example

Comment (andrewazores) [reply src/test/java/itest/GraphQLIT.java:789]: You can, yes, but the point of this class is to exercise the GraphQL endpoint, and this test in particular should exercise the new code you have added. If you use `GET /api/v1/recordings` you will get data back, but you will not be using the GraphQL `names` query that you added, so what is being tested exactly?

Comment (aali309) [reply src/test/java/itest/GraphQLIT.java:789]: ok. got it

File: src/test/java/itest/GraphQLIT.java (hunk 6 above)
Comment (aali309) [reply src/test/java/itest/GraphQLIT.java:648]: in my thinking then we do not need to create any active recordings? just retrieve any that are present then if any that are present, based on the data I will filter by names? The created recordings with the code that I have cannot be retrieved with ``` /api/v2.2/graphql ```. its always returning 0 for active recordings available

Comment (aali309) [reply src/test/java/itest/GraphQLIT.java:648]: never mind i think I saw something useful by creating a snapshot with ``` POST /api/v2/targets/:targetId/snapshot ``` this should work

Comment (andrewazores) [reply src/test/java/itest/GraphQLIT.java:648]: If you do not create any active recordings, there will be none there when you query and therefore none to filter...

> The created recordings with the code that I have cannot be retrieved with /api/v2.2/graphql. its always returning 0 for active recordings available

Then something is wrong, because these things should work together. Either there is a bug or your test query setup isn't working how you intended.

> never mind i think I saw something useful by creating a snapshot with POST /api/v2/targets/:targetId/snapshot this should work

I don't think so - creating a snapshot when there are no active recordings does not do anything.

Comment (aali309) [reply src/test/java/itest/GraphQLIT.java:648]: I might need help tomorrow if by any chance you have time. I think the problem is my graphql query. I read the .md files and ```"/api/v2.2/graphql"``` should be able to retrieve the created recordings

Comment (andrewazores) [reply src/test/java/itest/GraphQLIT.java:648]: Sure, just ping me or send a calendar invite.

Comment (aali309) [reply src/test/java/itest/GraphQLIT.java:648]: I will ping you... and when you are free we can meet 
mean while this is what i have 
``` CompletableFuture<TargetNodesQueryResponse> resp2 = new CompletableFuture<>();
 String query =
     "query { targetNodes {"
             + "name"
             + "nodeType"
             + "labels"
             + "target {"
             + "alias"
             + "serviceUri"
             + "}"
             + "recordings {"
             + "active {"
             + "data {"
             + "name"
             + "state"
             + "metadata {"
             + "labels"
             + "}"
             + "}"
             + "aggregate {"
             + "count"
             + "}"
             + "}"
             + "}"
             + "}"
             + "}";
 webClient.post("/api/v2.2/graphql")
     .sendJson(new JsonObject().put("query", query), ar -> {
         if (assertRequestStatus(ar, resp2)) {
             resp2.complete(
                     gson.fromJson(
                             ar.result().bodyAsString(),
                             TargetNodesQueryResponse.class));
         }
     });
 
     TargetNodesQueryResponse graphqlResp = resp2.get(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);``` trying to get all active recordings... i created two named Recording1 and Recording2

File: src/test/java/itest/GraphQLIT.java [hunk 8]
```diff
@@ -835,6 +1235,7 @@ public boolean equals(Object obj) {
     }
 
     static class TargetNodes {
+        public List<EnvironmentNode> environmentNodes;
```
[the remark this answers is not in the document]
Comment (aali309) [reply src/test/java/itest/GraphQLIT.java]: with the changes no, So i removed

File: src/test/java/itest/GraphQLIT.java [hunk 9]
```diff
@@ -588,6 +602,475 @@ void testNodesHaveIds() throws Exception {
...
+                .get(String.format("/api/v1/targets/%s/recordings", SELF_REFERENCE_TARGET_ID))
+                .send(
+                        ar -> {
+                            if (assertRequestStatus(ar, listRespFuture1)) {
+                                listRespFuture1.complete(ar.result().bodyAsJsonArray());
+                            }
+                        });
+        JsonArray listResp = listRespFuture1.get(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+        Assertions.assertTrue(listResp.isEmpty());
+        CountDownLatch latch = new CountDownLatch(2);
```
Comment (andrewazores) [inline src/test/java/itest/GraphQLIT.java:661]: I don't think this latch is necessary. Just waiting on the two `CompletableFuture`s in order will do the same thing as what this latch appears to do.

File: src/test/java/itest/GraphQLIT.java [hunk 10]
```diff
@@ -588,6 +602,475 @@ void testNodesHaveIds() throws Exception {
...
+
+        MatcherAssert.assertThat(
+                "list should have size 2 after recording creation",
+                listResp.size(),
+                Matchers.equalTo(2));
+
+        // GraphQL Query to filter Active recordings by names
+        CompletableFuture<TargetNodesQueryResponse> resp2 = new CompletableFuture<>();
+        String query =
+                "query { targetNodes {"
```
Comment (andrewazores) [inline src/test/java/itest/GraphQLIT.java:718]: Why not do `targetNodes(filter: { name: "some-node-name" })` here? This way the query looks only for active recordings on the specified target node. I think you could then remove the previous `/api/v1/targets/%s/recordings` query that checks the same data that this query does, and you would not really need the `for (TargetNode targetNode : graphqlResp.data.targetNodes)` loop lower down either.

File: src/test/java/itest/GraphQLIT.java [hunk 11]
```diff
@@ -588,6 +602,475 @@ void testNodesHaveIds() throws Exception {
...
+                        + "recordings {"
+                        + "active(filter: { names: [\"Recording1\", \"Recording2\"] }) {"
```
Comment (andrewazores) [inline src/test/java/itest/GraphQLIT.java:720]: If two recordings are created, and this filters for both of their expected values, then we don't know if this filter actually does anything - it might just pass through all results. Should this test perhaps query for only one of them? Or query with a filter that matches neither and assert that the response contains no entries? etc.

File: src/test/java/itest/GraphQLIT.java [hunk 12]
```diff
@@ -588,6 +602,475 @@ void testNodesHaveIds() throws Exception {
...
+        CompletableFuture<TargetNodesQueryResponse> resp2 = new CompletableFuture<>();
+        // List<String> filterNames = Arrays.asList("Recording1", "Recording2");
+        String query =
+                "query { targetNodes {"
+                        + "recordings {"
+                        + "archived(filter: { names: [\"io-cryostat-Cryostat_"
+                        + TEST_RECORDING_NAME
+                        + "_\\\\d{8}T\\\\d{6}Z(\\\\.\\\\d+)?\\\\.jfr\",\"io-cryostat-Cryostat_"
+                        + TEST_RECORDING_NAME
+                        + "_\\\\d{8}T\\\\d{6}Z(\\\\.\\\\d+)?\\\\.jfr\"] }) {"
```
Comment (andrewazores) [inline src/test/java/itest/GraphQLIT.java:887]: How does this work? I thought the implementation was looking for exact string matches for the recording names, but this test is passing a regular expression pattern as the recording name and expecting the backend to perform a regex match during the query. Where is this regular expression getting compiled and matched against... ?

File: src/test/java/itest/GraphQLIT.java (hunk 11 above)
Comment (aali309) [reply src/test/java/itest/GraphQLIT.java:720]: added one extra that will not be filtered

File: src/test/java/itest/GraphQLIT.java (hunk 10 above)
Comment (aali309) [reply src/test/java/itest/GraphQLIT.java]: I think that will be filtering node name instead of recording name? Also when I do that, it returns nothing with the logic of creating active recording that I used.

Comment (andrewazores) [reply src/test/java/itest/GraphQLIT.java]: That's right, it will filter the nodes, which you can do first and then on the result filter the recording name. The point would be that this way you only get the query response from that particular node - the one on which you started your test recordings - rather than a query response of all of the recordings on every node.

File: src/test/java/itest/GraphQLIT.java [hunk 13]
```diff
@@ -828,63 +801,51 @@ public void shouldReturnArchivedRecordingsFilteredByNames() throws Exception {
...
+
         String query =
                 "query { targetNodes {"
                         + "recordings {"
-                        + "archived(filter: { names: [\"io-cryostat-Cryostat_"
-                        + TEST_RECORDING_NAME
-                        + "_\\\\d{8}T\\\\d{6}Z(\\\\.\\\\d+)?\\\\.jfr\",\"io-cryostat-Cryostat_"
-                        + TEST_RECORDING_NAME
-                        + "_\\\\d{8}T\\\\d{6}Z(\\\\.\\\\d+)?\\\\.jfr\"] }) {"
+                        + "archived(filter: { names: [\"archivedRecording\",\"someOtherName\"] }) {"
```
Comment (andrewazores) [inline src/test/java/itest/GraphQLIT.java:848]: I don't think the archived file names will be like this, they would look like `io-cryostat-Cryostat_some-recording-name_12345678.jfr`, right?

I think the way to test this would be to use a different query, like `GET /api/v1/recordings`, to get the list of all of the archived recordings within the test. Then, within the test, you can apply some logic to filter out the ones that match your expectations, ie the `some-recording-name` matches the name you gave the recordings. Then you use that information to build the GraphQL query filter to pass in here and ensure that the GraphQL response also matches your expectation.

Comment (aali309) [reply src/test/java/itest/GraphQLIT.java:848]: yes the name expected should be the way you described

Comment (aali309) [reply src/test/java/itest/GraphQLIT.java:849]: I used ```GET /api/v1/recordings``` and then extracted the name of the archived recording to a variable1 then I used ```.post("/api/v2.2/graphql")``` and extraxted the archived recording to a variable2 then checked if they match.

File: src/main/java/io/cryostat/net/web/http/api/v2/graph/ActiveRecordingsFetcher.java [hunk 14]
```diff
@@ -97,9 +97,17 @@ public Active getAuthenticated(DataFetchingEnvironment environment) throws Excep
             String recordingName = filter.get(FilterInput.Key.NAME);
             recordings =
                     recordings.stream()
-                            .filter(r -> Objects.equals(r.getName(), recordingName))
+                            .filter(r -> r.getName().equals(recordingName))
```
Comment (andrewazores) [inline src/main/java/io/cryostat/net/web/http/api/v2/graph/ActiveRecordingsFetcher.java:100]: Any reason to change this `Objects.equals()` call?

Comment (aali309) [reply src/main/java/io/cryostat/net/web/http/api/v2/graph/ActiveRecordingsFetcher.java:100]: just following the consistency of all other classes so they behave the same

File: src/test/java/itest/GraphQLIT.java [hunk 15]
```diff
@@ -588,6 +601,426 @@ void testNodesHaveIds() throws Exception {
...
+        List<String> filterNames = Arrays.asList("Recording1", "Recording2");
+
+        for (TargetNode targetNode : graphqlResp.data.targetNodes) {
+            List<ActiveRecording> recordings = targetNode.recordings.active.data;
+            for (ActiveRecording recording : recordings) {
+                if (filterNames.contains(recording.name)) {
+                    filteredRecordings.add(recording);
+                }
+            }
+        }
```
Comment (andrewazores) [inline src/test/java/itest/GraphQLIT.java:726]: What is this for-loop needed for? It seems like it is providing a filtered version of the query response that only contains the expected data, and then the test continues to assert that this filtered version matches the expectations - but it has to, because this loop is trimming the response down to the expectation first. I think the assertions below can just be done directly on the query response, can't they?

File: src/test/java/itest/GraphQLIT.java [hunk 16]
```diff
@@ -588,6 +601,426 @@ void testNodesHaveIds() throws Exception {
...
+                            } else {
+
+                                archiveRecordingFuture.completeExceptionally(
+                                        new RuntimeException("Archive request failed"));
+                            }
+                        });
+
+        archiveRecordingFuture.get(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+
+        Thread.sleep(2_000L);
```
Comment (andrewazores) [inline src/test/java/itest/GraphQLIT.java:826]: I think you probably want to put a sleep in between creating the active recording and archiving that recording, so that there is time for the recording to gather data before it is copied to archives. Putting the sleep here between archiving and querying for what archives exist is probably not very useful.

File: src/test/java/itest/GraphQLIT.java (hunk 13 above)
Comment (andrewazores) [reply src/test/java/itest/GraphQLIT.java:849]: I'm not sure that this is resolved - the GraphQL query string here still asks the server for `"archivedRecording" and `"someOtherName"`, but those still shouldn't be the actual filenames of archived recordings. I thought that the solution would look like `archived(filter: { names: [" + retrievedArchivedRecordingsName + "] } )`

File: src/main/java/io/cryostat/net/web/http/api/v2/graph/ActiveRecordingsFetcher.java (hunk 14 above)
Comment (andrewazores) [reply src/main/java/io/cryostat/net/web/http/api/v2/graph/ActiveRecordingsFetcher.java:100]: Do the others not use `Objects.equals()`? How does `Objects.equals(r.getName(), recordingName))` behave differently from `r.getName().equals(recordingName)`?

File: src/test/java/itest/GraphQLIT.java (hunk 16 above)
Comment (aali309) [reply src/test/java/itest/GraphQLIT.java:826]: I erased a big chunk of code that was retrieving to make sure the recording is archived... forgot to remove the sleep

Comment (andrewazores) [reply src/test/java/itest/GraphQLIT.java:826]: I think the sleep makes sense to keep, but it should be in between creating the new active recording and archiving it.

File: src/test/java/itest/GraphQLIT.java (hunk 13 above)
Comment (aali309) [reply src/test/java/itest/GraphQLIT.java]: this works too instead of hard cording the name

File: src/test/java/itest/GraphQLIT.java (hunk 15 above)
Comment (aali309) [reply src/test/java/itest/GraphQLIT.java]: I got rid of the for loop and assert directly

File: src/test/java/itest/GraphQLIT.java (hunk 16 above)
Comment (aali309) [reply src/test/java/itest/GraphQLIT.java]: yes

**Review by andrewazores — APPROVED** [review]

**Comment by andrewazores** [comment]
Looks good! Please rebase and I will merge this.

**Review by andrewazores — APPROVED** [review]
